### PR TITLE
[MOV] (_ => sale_)loyalty_delivery: merge modules

### PR DESCRIPTION
**Pull request by Feyensv**
There is no need to have two bridges linking delivery & loyalty. The two modules held the two parts of the same logic, making no sense to keep the two modules separate.

Task-3420579

Upgrade PR: https://github.com/odoo/upgrade/pull/4987

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_loyalty_delivery/models/sale_order.py [hunk 1]
```diff
@@ -4,15 +4,26 @@
...
 
+    # delivery overrides
+
+    def _compute_amount_total_without_delivery(self):
+        res = super()._compute_amount_total_without_delivery()
+        return res - sum(
+            self.order_line.filtered(
+                lambda l: l.coupon_id and l.coupon_id.program_type in ['ewallet', 'gift_card']
+            ).mapped('price_unit')
+        )
```
Comment (vava-odoo) [inline addons/sale_loyalty_delivery/models/sale_order.py:19]: Nitpick, but one line less
```suggestion
        return res - sum(self.order_line.filtered(
            lambda l: l.coupon_id and l.coupon_id.program_type in ['ewallet', 'gift_card']
        ).mapped('price_unit')
```

File: addons/sale_loyalty_delivery/models/sale_order.py [hunk 2]
```diff
@@ -4,15 +4,26 @@
...
+
+    # sale_loyalty overrides
+
     def _get_no_effect_on_threshold_lines(self):
-        self.ensure_one()
```
Comment (vava-odoo) [inline addons/sale_loyalty_delivery/models/sale_order.py:11]: I don't mind that you remove this from overrides, but please apply it to all overrides of this file then (`_get_reward_line_values`)

**Review by vava-odoo — APPROVED** [review]
Thanks @Feyensv!
I'm sure you would have asked me to split the commit (one for cleaning, and one for moving), but I'll close my eyes on that one :innocent:

File: addons/sale_loyalty_delivery/models/sale_order.py (hunk 1 above)
Comment (Feyensv) [reply addons/sale_loyalty_delivery/models/sale_order.py:19]: Not convinced, it's less clear imho.

File: addons/sale_loyalty_delivery/models/sale_order.py (hunk 2 above)
Comment (Feyensv) [reply addons/sale_loyalty_delivery/models/sale_order.py:11]: Only removing it when the super call does the check and can be called first, which isn't the case in `_get_reward_line_values`

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**